tests/cupyx_tests/scipy_tests/fft_tests/test_realtransforms.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+import pytest
+
+import cupyx.scipy.fft  # noqa
+from cupyx.scipy import fft as cp_fft
+from cupy import testing
+
+try:
+    # scipy.fft is available since scipy v1.4.0+
+    import scipy.fft as scipy_fft  # noqa
+except ImportError:
+    scipy_fft = None
+    pass
+
+# used to avoid SciPy bug in complex dtype cases with output_x=True
+# https://github.com/scipy/scipy/pull/11904
+scipy_cplx_bug = not cupyx.scipy.fft._scipy_150
+
+if cupyx.scipy.fft._scipy_160:
+    # additional normalization options available for SciPy>=1.6
+    all_dct_norms = [None, 'ortho', 'forward', 'backward']
+else:
+    all_dct_norms = [None, 'ortho']
+
+
+@testing.parameterize(
+    *testing.product(
+        {
+            'n': [None, 0, 5, 15],
+            'type': [1, 2, 3, 4],
+            'shape': [(9,), (10,), (10, 9)],
+            'axis': [-1, 0],
+            'norm': ['ortho'],
+            'overwrite_x': [False],
+            'function': ['dct', 'dst', 'idct', 'idst'],
+        }
+    )
+    # test all overwrite_x and norm combinations on a smaller subset of shapes
+    + testing.product(
+        {
+            'n': [None, 15],
+            'type': [2, 3],
+            'shape': [(10, 9)],
+            'axis': [-1, 0],
+            'norm': all_dct_norms,
+            'overwrite_x': [False, True],
+            'function': ['dct', 'dst', 'idct', 'idst'],
+        }
+    )
+
+)
+@testing.gpu
+@testing.with_requires('scipy>=1.4')
+class TestDctDst():
+
+    def _run_transform(self, dct_func, xp, dtype):
+        x = testing.shaped_random(self.shape, xp, dtype)
+        if scipy_cplx_bug and x.dtype.kind == 'c':
+            # skip cases where SciPy has a bug
+            return x
+        x_orig = x.copy()
+        kwargs = dict(type=self.type,
+                      n=self.n,
+                      axis=self.axis,
+                      norm=self.norm,
+                      overwrite_x=self.overwrite_x)
+        if self.type in [1, 4]:
+            if xp != np:
+                # type 1 and 4 real-to-real transforms not implemented
+                with pytest.raises(NotImplementedError):
+                    dct_func(x, **kwargs)
+            return xp.zeros([])
```

Comment:
Can't we just return/skip (not even calling SciPy) for type 1 & 4?
```suggestion
        if self.type in (1, 4):
            # type 1 and 4 real-to-real transforms not implemented
            return xp.zeros([])
```

Reply:
The intention was to test that the NotImplementedError was being raised, but that will only occur for CuPy since those types are available in SciPy.

Reply:
I wouldn't test if something is not yet there, but I suppose it's ok to keep as is, allowing simple changes when they are added in the future 🙂

Reply:
I am fine with removing this and removing type 1 and 4 from the parametrization if you prefer that

Reply:
I am fine with that, let's wait for @takagi's opinion.

Reply:
I want to keep type 1 and 4 in the combination, and either is fine checking `NotImplementedError` or not. Let's leave it as it is this time.